File: app/src/main/java/com/bykea/pk/partner/widgets/AutoFitFontTextView.java
package com.bykea.pk.partner.widgets;

import android.content.Context;
import android.content.res.TypedArray;
import android.text.TextUtils;
import android.util.AttributeSet;
import android.widget.TextView;

import com.bykea.pk.partner.R;

import me.grantland.widget.AutofitTextView;

public class AutoFitFontTextView extends AutofitTextView {

    public AutoFitFontTextView(Context context) {
        super(context);
    }

    public AutoFitFontTextView(Context context, AttributeSet attrs) {
        super(context, attrs);
        applyAttributes(context, attrs);
    }

    public AutoFitFontTextView(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
        applyAttributes(context, attrs);
    }

    private void applyAttributes(Context context, AttributeSet attrs) {
        TypedArray a = context.obtainStyledAttributes(attrs,
                R.styleable.FontTextView);
        String fontName = Fonts.values()[a.getInt(R.styleable.FontTextView_fontName, 0)].getName();
        a.recycle();
        if (!isInEditMode() && !TextUtils.isEmpty(fontName)) {
            setTypeface(FontUtils.getFonts(getContext(), fontName));
        }
    }
}

File: loadboard/src/main/java/com/bykea/pk/partner/loadboard/widgets/AutoFitFontTextView.kt
package com.bykea.pk.partner.loadboard.widgets

import android.content.Context
import android.text.TextUtils
import android.util.AttributeSet

import com.bykea.pk.partner.loadboard.R
import me.grantland.widget.AutofitTextView

class AutoFitFontTextView : AutofitTextView {

    constructor(context: Context) : super(context) {}

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        applyAttributes(context, attrs)
    }

    constructor(context: Context, attrs: AttributeSet, defStyle: Int) : super(context, attrs, defStyle) {
        applyAttributes(context, attrs)
    }

    private fun applyAttributes(context: Context, attrs: AttributeSet) {
        val a = context.obtainStyledAttributes(attrs,
                R.styleable.FontTextView)
        val fontName = Fonts.values()[a.getInt(R.styleable.FontTextView_fontName, 0)].getName()
        a.recycle()
        if (!isInEditMode && !TextUtils.isEmpty(fontName)) {
            typeface = FontUtils.getFonts(context,fontName)
        }
    }
}